src/path_functions.cpp

```diff
@@ -44,107 +44,76 @@ QString AOApplication::get_data_path() { return get_base_path() + "data/"; }
 QString AOApplication::get_default_theme_path(QString p_file)
 {
   QString path = get_base_path() + "themes/default/" + p_file;
-#ifndef CASE_SENSITIVE_FILESYSTEM
-  return path;
-#else
   return get_case_sensitive_path(path);
-#endif
 }
 
 QString AOApplication::get_custom_theme_path(QString p_theme, QString p_file)
 {
   QString path = get_base_path() + "themes/" + p_theme + "/" + p_file;
-#ifndef CASE_SENSITIVE_FILESYSTEM
-  return path;
-#else
   return get_case_sensitive_path(path);
-#endif
 }
 
 QString AOApplication::get_theme_path(QString p_file)
 {
   QString path = get_base_path() + "themes/" + current_theme + "/" + p_file;
-#ifndef CASE_SENSITIVE_FILESYSTEM
-  return path;
-#else
   return get_case_sensitive_path(path);
-#endif
 }
 
 QString AOApplication::get_character_path(QString p_char, QString p_file)
 {
   QString path = get_base_path() + "characters/" + p_char + "/" + p_file;
-#ifndef CASE_SENSITIVE_FILESYSTEM
-  return path;
-#else
   return get_case_sensitive_path(path);
-#endif
 }
 
 QString AOApplication::get_sounds_path(QString p_file)
 {
   QString path = get_base_path() + "sounds/general/" + p_file;
-#ifndef CASE_SENSITIVE_FILESYSTEM
-  return path;
-#else
   return get_case_sensitive_path(path);
-#endif
 }
 
 QString AOApplication::get_music_path(QString p_song)
 {
   QString path = get_base_path() + "sounds/music/" + p_song;
-#ifndef CASE_SENSITIVE_FILESYSTEM
-  return path;
-#else
   return get_case_sensitive_path(path);
-#endif
 }
 
 QString AOApplication::get_background_path(QString p_file)
 {
   QString path = get_base_path() + "background/" +
                  w_courtroom->get_current_background() + "/" + p_file;
   if (courtroom_constructed) {
-#ifndef CASE_SENSITIVE_FILESYSTEM
-    return path;
-#else
     return get_case_sensitive_path(path);
-#endif
   }
   return get_default_background_path(p_file);
 }
 
 QString AOApplication::get_default_background_path(QString p_file)
 {
   QString path = get_base_path() + "background/default/" + p_file;
-#ifndef CASE_SENSITIVE_FILESYSTEM
-  return path;
-#else
   return get_case_sensitive_path(path);
-#endif
 }
 
 QString AOApplication::get_evidence_path(QString p_file)
 {
   QString path = get_base_path() + "evidence/" + p_file;
-#ifndef CASE_SENSITIVE_FILESYSTEM
-  return path;
-#else
   return get_case_sensitive_path(path);
-#endif
 }
 
 QString AOApplication::get_case_sensitive_path(QString p_file)
 {
+  QFileInfo file(p_file);
+  QString file_basename = file.fileName();
+
+  // no path traversal above base folder
+  if (!(file.absolutePath().startsWith(get_base_path())))
```

Comment:
Does `absolutePath` follow symlinks? If it does, it might break use of symbolic links that point somewhere outside the base folder.

Reply:
it does not, so your symlink hacks will keep working
only canonicalPath() will follow them and it returns empty strings on windows